src/include/daos/common.h

```diff
@@ -833,4 +833,14 @@ daos_anchor_is_zero(daos_anchor_t *anchor)
 /* default debug log file */
 #define DAOS_LOG_DEFAULT	"/tmp/daos.log"
 
+/* Secure memory scrub */
+#define barrier_data(ptr) \
+	asm volatile("" :  : "r"(ptr) : "memory")
+
+static inline void
```

Comment:
Have you thought about writing something invalid here, rather than 0, to make it harder to recover the bits that were there before?  I know that normally, clearing the memory is logically correct.  I thought there are attacks that can recover data from buffers that were zeroed though (under the right circumstances).

Reply:
I used the implementation in the SDL task so I guess they think this is sufficient. I think the restoring data that isn't overwritten with random data is for hard drives not so much memory. The most important part of this patch is that the variable is marked volatile so it won't get compiled out and that there is a barrier to ensure the write gets flushed from the cache which is probably where those other attacks come from.